packages/platform-core/src/backend_identifier_parts_conversions.ts

```diff
@@ -0,0 +1,70 @@
+import { BackendIdentifierParts } from '@aws-amplify/plugin-types';
+
+const STACK_NAME_LENGTH_LIMIT = 128;
+const AMPLIFY_PREFIX = 'amplify';
+const NUM_DASHES = 3;
+
+/**
+ * Serialize BackendIdentifierParts into a CFN stack name.
+ *
+ * Note that this is a LOSSY conversion.
+ * Some disallowed characters are removed from the BackendIdentifierParts when converting to a stack name and long
+ * values are truncated to fit in the length constraints
+ */
+export const backendIdentifierPartsToStackName = (
```

Comment:
I started down this route, but changed course because the `BackendIdentifier` ends up in the API of the `deployed-backend-client` and `client-config` which we want to be pure data-driven APIs. If we have a class with subtypes then calling the methods in `deployed-backend-client` becomes a hunt to find where the impls that you can pass into the methods are defined. I would prefer to keep `BackendIdentifier` as a pure data type and have the transformations be a separate class that operates on that data.

Reply:
Gotcha, we can then do what Kamil suggested, create a converter/marshaller or even (`BackendIdentifierFactory` and `StackNameFactory`).

Reply:
My vote would be for a single converter just to avoid factory explosion.

Reply:
Me too. It's bidirectional flow that conversion. So converter/serializer/marshaller seems to be better naming.
Factory is usually one direction.